timesketch/api/v1/resources/sigma.py

```diff
@@ -153,11 +213,374 @@ def post(self):
+        return_rule = []
+
+        assert isinstance(rule, SigmaRule)
+
+        return_rule.append(_enrich_sigma_rule_object(rule=rule))
+
+        return jsonify({"objects": return_rule, "meta": {}})
+
+    @login_required
+    def delete(self, rule_uuid):
+        """Handles DELETE request to the resource.
+
+        Handels DELETE API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+        Deletes the SigmaRule matching the specified UUID from the database.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            HTTP_STATUS_CODE_NOT_FOUND if rule not found
+            HTTP_STATUS_CODE_OK if rule deleted
+        """
+
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        if not rule:
+            error_msg = "No rule found with rule_uuid.{0!s}".format(rule_uuid)
+            logger.debug(error_msg)  # only needed in debug cases
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                error_msg,
+            )
+
+        db_session.delete(rule)
+        db_session.commit()
+
+        return HTTP_STATUS_CODE_OK
+
+    @login_required
+    def post(self, rule_uuid=None):
+        """Handles POST request to the resource.
+
+        Handels POST API calls to /sigmarule/ where the
+        rule_uuid is the primaray way to identify the rule.
+
+        The rule is created in the database if non with the given
+        rule_uuid is found in the database
+
+        The remaining attributes of the rule are provided in request itself.
+
+        If no `rule_yaml` is found in the reuqest, the method will fail as this
+        is required to parse the rule.
+
+        In case no `rule_uuid`is given as a parameter, the methd will try to
+        get it from the yaml file.
+
+        If the `rule_uuid` parameter is different to the one provided in the
+        yaml file, the one from the yaml file is used.
+
+        To update a rule, use `PUT`instead.
+
+        Args:
+            rule_uuid: optional field that is not needed
+
+        Returns:
+            Sigma rule object and HTTP status code indicating
+            whether operation was sucessful.
+        """
+        rule_yaml = request.json.get("rule_yaml")
+
+        if not rule_yaml:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "No rule_yaml supplied in the POST request to the API.",
+            )
+
+        if not isinstance(rule_yaml, str):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN, "rule_yaml needs to be a string."
+            )
+        try:
+            parsed_rule = ts_sigma_lib.parse_sigma_rule_by_text(rule_yaml)
+        except ValueError as e:
+            error_msg = "Sigma Rule Parsing error: {0!s}".format(e)
+            logger.debug(error_msg)
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
```

Comment:
Isn't this the same situation than in L 371? Essentially, you just want to make sure that the rule is parseable. SO maybe the check in 369 is not necessary here?